presto-main/src/main/java/io/prestosql/sql/planner/planPrinter/IoPlanPrinter.java

```diff
@@ -61,11 +66,21 @@
 import static io.airlift.json.JsonCodec.jsonCodec;
 import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
 import static io.prestosql.spi.predicate.Marker.Bound.EXACTLY;
+import static io.prestosql.sql.planner.planPrinter.PlanPrinterUtil.castToVarchar;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class IoPlanPrinter
 {
+    /**
+     * List of supported data types for column constraints appearing in
+     * EXPLAIN (TYPE IO). Other types would cause a NOT_SUPPORTED
+     * Presto exception to be thrown.
+     */
+    private static final Set<Class> WHITELISTED_TYPES_FOR_OUTPUT = ImmutableSet.of(
```

Comment:
We could support any type that has a proper CAST to VARCHAR instead of an explicit whitelist.

Reply:
So, the implementation of getVarcharValue() below should check if there's a CAST (like in PlanPrinter.castToVarchar) and invoke it or throw the "Unsupported data type in EXPLAIN (TYPE IO)" exception.

Reply:
Thanks for the feedback, @martint. I've pushed a new commit that removed the whitelist and instead relies on whether the CAST to VARCHAR succeeds or fails.